### PR TITLE
feat(check:trailing_whitespace):  collect all problems, report and then exit

### PR DESCRIPTION
**Pull request by rwaffen**
fixes: #140

### CODE REVIEW

File: lib/voxpupuli/test/rake.rb [hunk 1]
```diff
@@ -12,15 +12,22 @@
...
       File.foreach(filename).each_with_index do |line, index|
         if line =~ %r{\s\n$}
-          puts "#{filename} has trailing whitespace on line #{index + 1}"
-          exit 1
+          errors << "#{filename} has trailing whitespace on line #{index + 1}"
         end
       end
     end
+
+    unless errors.empty?
```
Comment (ekohl) [inline lib/voxpupuli/test/rake.rb:26]: I'm still hesitant on using `unless` when I can avoid it
```suggestion
    if errors.any?
```

**Review by ekohl — APPROVED** [review]
I can't quite get used to conventional commits. I'm not sure how this is for others. Wouldn't it be confusing to use it for some, but not all?

Other than that, the implementation looks correct. Just one tiny note.

**Comment by rwaffen** [comment]
I try to do conventional commits more and more, because in my project people also starting with it and we want somehow build semantic-release upon it. but I'm not really sure about it my self, thou. or like these young people say: "I don't feel the vibe" :D

**Comment by ekohl** [comment]
I'm also looking at [towncrier](https://github.com/twisted/towncrier) and that it provides a bit more flexibility in its structure. Though you have more "noise" in your git log for files. One advantage is that with GitHub (& GitLab for that matter) you can't comment on commit messages, making it very hard to provide feedback on them. Even if it's for typos.

**Pull request merged**